gramps/gen/lib/grampstype.py

```diff
@@ -165,11 +179,13 @@ def __set_tuple(self, value):
         self.__value = val
         self.__string = strg
 
-    def __set_int(self, value):
+    @set.register
+    def __set_int(self, value: int):
         "Set the value/string properties from an integer."
         self.__value = value
         self.__string = ""
 
+    # This method needs to be registered outside of the class.
     def __set_instance(self, value):
```

Comment:
I would still type hint this as well. Note, if you want to use `GrampsType` directly, you would have to add `from __future__ import annotations`, though you can also quote it instead.

Reply:
I've added the type hint back using this approach.
I could not find a way to move the register back into the class so it is left at the end of the file